### ISSUE TITLE
Large number of warnings being raised by `RemoveAssetFromDatabase`

### ISSUE THREAD

**Issue by rhys-vdw** [ogxd/project-curator#16]
We're frequently getting these warnings from this project.

```
Asset 'Assets/_Game/Content/Chronospheres/ChronosphereUtilities/BiomeEquipmentTables/ClockworkTables/ClockworkItemTables/ClockworkItemTable.asset' that depends on 'Assets/_Game/Content/Clockwork/ClockworkItems/PowerCycle/PowerCycle.asset' doesn't have it as a dependency
UnityEngine.Debug:LogWarning (object)
Ogxd.ProjectCurator.ProjectCurator:RemoveAssetFromDatabase (string) (at ./Library/PackageCache/com.ogxd.project-curator@5a5eb9a295/Editor/ProjectCurator.cs:66)
Ogxd.ProjectCurator.AssetProcessor/<>c__DisplayClass3_0:<OnWillSaveAssets>b__0 () (at ./Library/PackageCache/com.ogxd.project-curator@5a5eb9a295/Editor/AssetProcessor.cs:38)
Ogxd.ProjectCurator.AssetProcessor:OnUpdate () (at ./Library/PackageCache/com.ogxd.project-curator@5a5eb9a295/Editor/AssetProcessor.cs:25)
UnityEditor.EditorApplication:Internal_CallUpdateFunctions ()
```

Is there some actionable fix that should be made here? Otherwise I'd suggest only enabling this warning with a compiler symbol like `PROJECT_CURATOR_DEBUG`, as I'm not sure what we should do about it at our end.

Happy to contribute this change.

**Comment by rhys-vdw** [comment]
A particular class of this warning (when moving/renaming files) is fixed by #17. However I still see these on database rebuilt.

**Comment by rhys-vdw** [comment]
~~I'm pretty sure that `OnWillSaveAssets` should either do nothing, or, if for some reason the asset is not yet in the database, add it.~~

~~There is no need for this Remove -> Add combo (which is causing the warning spam).~~

Ah nvm I'm mistaken. It does need to update the deps of course.

**Comment by rhys-vdw** [comment]
Pretty sure the last of these I was seeing were from folders, so it should be closed by #23

**Comment by rhys-vdw** [comment]
This issue is ongoing btw, can it be reopened @ogxd?

IIRC it's caused by logic error:

https://github.com/ogxd/project-curator/blob/master/Editor/AssetProcessor.cs#L39-L40

This logic inevitably results in the warning (since it's removing and adding the same asset). The warning is just wrong for this case I'm pretty sure.
